tests/devices_test/dependencies_test.py

```diff
@@ -129,3 +134,111 @@ def test_dependencies_handling(self):
         self.assertFalse(fmt.supported)
         self.assertFalse(fmt.formattable)
         self.assertFalse(fmt.destroyable)
+
+
+class MissingWeakDependenciesTestCase(unittest.TestCase):
+    def setUp(self):
+        self.addCleanup(self._clean_up)
+        self.disk1_file = create_sparse_tempfile("disk1", Size("2GiB"))
+
+    def _clean_up(self):
+        # reload all libblockdev plugins
+        blivet.blockdev.try_reinit(require_plugins=None, reload=False)
+
+        for disk in self.bvt.disks:
+            self.bvt.recursive_remove(disk)
+
+        self.bvt.reset()
```

Comment:
I don't think you want this `reset` call. It will populate the tree again. Maybe you wanted `self.bvt.devicetree.teardown_all`?